app/controllers/buses_controller.rb

```diff
@@ -0,0 +1,53 @@
+require 'net/http'
+require 'json'
+
+class BusesController < ApplicationController
+  def index
+  end
+
+  def stop
+    stop = params[:stop]
+    @arrivals = fetch_arrivals([stop])
+    render :arrivals
+  end
+
+  def postcode
+    postcode = params[:postcode]
```

Comment:
Should we validate the postcode? What happens if the postcode is invalid?

Even if it doesn't crash and returns an empty list we should probably say something about it (some of this might be better to be added in the views)

Reply:
Now validating in `fetch_arrivals` and in the view